erpnext/accounts/doctype/payment_request/payment_request.py

```diff
@@ -214,9 +214,11 @@ def set_as_cancelled(self):
 
 	def check_if_payment_entry_exists(self):
 		if self.status == "Paid":
-			payment_entry = frappe.db.sql_list("""select parent from `tabPayment Entry Reference`
-				where reference_name=%s""", self.reference_name)
-			if payment_entry:
+			payment_entry = frappe.get_all("Payment Entry Reference",
+				filters={"reference_name": self.reference_name, "docstatus": ["<", 2]},
+				fields=["distinct(parent)"])
```

Comment:
TIL we can do this.
No need for distinct though, if it's only being used for validation (increases query time).

Reply:
Actually, `distinct` performs much better here. It's more readable, but marginally slower than `group by`.

![image](https://user-images.githubusercontent.com/13396535/53320480-d0a60a00-38fb-11e9-9f0f-161b3768ce29.png)